### PR TITLE
Added ability for NOPE to autoset cluster URL and token

### PR DESCRIPTION
**Pull request by nathan-weinberg**
Resolves NETOBSERV-331

### CODE REVIEW

File: scripts/nope.py [hunk 1]
```diff
@@ -91,19 +92,33 @@ def main():
...
-	parser.add_argument("--token", type=str, help='Token for Bearer auth', required=True)
+	parser.add_argument("--url", type=str, default='', help='Override for Cluster URL', required=False)
+	parser.add_argument("--token", type=str, default='', help='Override for Bearer auth token', required=False)
 	parser.add_argument("--namespace", type=str, default='network-observability', help='Namespace for query - defaults to "network-observability"', required=False)
 
 	# parse arguments
 	args = parser.parse_args()
+
 	CLUSTER_URL = args.url
+	if CLUSTER_URL == '':
```
Comment (memodi) [inline scripts/nope.py:103]: IMO, `if CLUSTER_URL is None` is much cleaner to check for empty strings.

File: scripts/nope.py [hunk 2]
```diff
@@ -91,19 +92,33 @@ def main():
 	parser = argparse.ArgumentParser(description='Network Observability Prometheus and Elasticsearch tool (NOPE)')
 
 	# set customization flags
-	parser.add_argument("--url", type=str, help='Cluster URL', required=True)
-	parser.add_argument("--token", type=str, help='Token for Bearer auth', required=True)
+	parser.add_argument("--url", type=str, default='', help='Override for Cluster URL', required=False)
+	parser.add_argument("--token", type=str, default='', help='Override for Bearer auth token', required=False)
```
Comment (memodi) [inline scripts/nope.py:96]: same comment as above.

**Review by memodi — CHANGES_REQUESTED** [review]

File: scripts/nope.py [hunk 3]
```diff
@@ -91,19 +92,33 @@ def main():
...
 	# parse arguments
 	args = parser.parse_args()
+
 	CLUSTER_URL = args.url
+	if CLUSTER_URL is None:
+		CLUSTER_URL = subprocess.run(['oc', 'get', 'infrastructure', 'cluster', '-o', 'jsonpath="{.status.apiServerURL}"'], capture_output=True, text=True).stdout
+		print(f"CLUSTER_URL (raw): {CLUSTER_URL}")
+
 	TOKEN = args.token
+	if TOKEN is None:
```
Comment (paigerube14) [inline scripts/nope.py:108]: Could also do `if not TOKEN:` but either works...

Comment (nathan-weinberg) [reply scripts/nope.py:108]: Yeah functionally they are the same - we could change it - do you think we should?

Comment (paigerube14) [reply scripts/nope.py:108]: nah think it's fine!

**Pull request merged**